src/pcap/bosh.go

```diff
@@ -1,55 +1,512 @@
+	return endpoints, nil
+}
+
+// matchesInstanceGroups determines whether the instance matches one of the selected groups.
+func matchesInstanceGroups(instance BoshInstance, groups []string) bool {
+	for _, validGroup := range groups {
+		if instance.Job == validGroup {
+			return true
+		}
+	}
+	return false
+}
+
+// matchesInstanceGroups determines whether the instance matches one of the selected instance IDs.
+func matchesInstanceIDs(instance BoshInstance, ids []string) bool {
+	for _, validID := range ids {
+		if instance.ID == validID {
+			return true
+		}
+	}
+	return false
+}
+
+func (br *BoshResolver) Validate(endpointRequest *EndpointRequest) error {
+	boshRequest := endpointRequest.GetBosh()
+
+	if boshRequest == nil {
+		return fmt.Errorf("invalid message: br: %w", errNilField)
 	}
 
-	if request.Token == "" {
+	if boshRequest.Token == "" {
 		return fmt.Errorf("invalid message: token: %w", errEmptyField)
 	}
 
-	if request.Deployment == "" {
+	if boshRequest.Deployment == "" {
 		return fmt.Errorf("invalid message: deployment: %w", errEmptyField)
 	}
 
-	if len(request.Groups) == 0 {
+	if boshRequest.Environment == "" {
+		return fmt.Errorf("invalid message: environment: %w", errEmptyField)
+	}
+
+	if len(boshRequest.Groups) == 0 {
 		return fmt.Errorf("invalid message: instance group(s): %w", errEmptyField)
 	}
 
 	return nil
 }
+
+func (br *BoshResolver) setup() error {
+	br.logger.Debug("setting up BoshResolver", zap.Any("resolver-config", br.Config))
+
+	var tlsConfig *tls.Config
+
+	if br.Config.MTLS != nil {
+		tlsConfig = &tls.Config{
+			MinVersion: tls.VersionTLS12,
+			MaxVersion: tls.VersionTLS13,
+			ClientAuth: tls.RequireAndVerifyClientCert,
+			RootCAs:    br.boshRootCAs,
+		}
+	}
+
+	br.client = &http.Client{
+		Transport: &http.Transport{
+			DialContext: (&net.Dialer{
+				Timeout: 500 * time.Millisecond, //nolint:gomnd // Default configuration
+			}).DialContext,
+			TLSHandshakeTimeout:   500 * time.Millisecond, //nolint:gomnd // Default configuration
+			ResponseHeaderTimeout: 500 * time.Millisecond, //nolint:gomnd // Default configuration
+			ExpectContinueTimeout: 500 * time.Millisecond, //nolint:gomnd // Default configuration
+			DisableKeepAlives:     true,
+			MaxIdleConnsPerHost:   -1,
+			TLSClientConfig:       tlsConfig,
+		},
+		Timeout: time.Second,
+	}
+
+	br.logger.Debug("discovering bosh-UAA endpoint", zap.String("bosh-director", br.DirectorURL.String()))
+	apiResponse, err := br.info()
+	if err != nil {
+		return err
+	}
+
+	br.UaaURLs = apiResponse.UserAuthentication.Options.URLs
+	br.logger.Info("connected to bosh-director", zap.Any("bosh-director", br.DirectorURL.String()))
+	return nil
+}
+
+// info retrieves the BOSH director /info endpoint.
+//
+// Used for startup and health check.
+func (br *BoshResolver) info() (*BoshInfo, error) {
+	// FIXME: Workaround for URL.JoinPath, which is buggy: https://github.com/golang/go/issues/58605
```

Comment:
Does it work if we set the initial path after parsing to `/`?

Reply:
It does.